src/pkg/cli/composeStart.go

```diff
@@ -163,27 +142,52 @@ func convertServices(ctx context.Context, c client.Client, serviceConfigs compos
 		network := network(&svccfg)
 		ports := convertPorts(svccfg.Ports)
 		services = append(services, &defangv1.Service{
-			Name:        NormalizeServiceName(svccfg.Name),
-			Image:       svccfg.Image,
-			Build:       build,
-			Internal:    network == defangv1.Network_PRIVATE,
-			Networks:    network,
-			Init:        init,
-			Ports:       ports,
-			Healthcheck: healthcheck,
-			Deploy:      deploy,
-			Environment: envs,
-			Secrets:     secrets,
-			Command:     svccfg.Command,
-			Domainname:  svccfg.DomainName,
-			Platform:    convertPlatform(svccfg.Platform),
-			DnsRole:     dnsRole,
-			StaticFiles: staticFiles,
+			Name:            NormalizeServiceName(svccfg.Name),
+			Image:           svccfg.Image,
+			Build:           build,
+			Internal:        network == defangv1.Network_PRIVATE,
+			Networks:        network,
+			Init:            init,
+			Ports:           ports,
+			Healthcheck:     healthcheck,
+			Deploy:          deploy,
+			Command:         svccfg.Command,
+			Domainname:      svccfg.DomainName,
+			Platform:        convertPlatform(svccfg.Platform),
+			DnsRole:         dnsRole,
+			StaticFiles:     staticFiles,
+			EnvironmentVars: envVars,
```

Comment:
Environment and Secrets fields are removed and EnvironmentVars are added to contain both.